File: restclient-classic/src/main/java/com/example/demo/Post.java
package com.example.demo;

import java.io.Serializable;
import java.time.LocalDateTime;

public record Post(String id,
                   String title,
                   String content,
                   LocalDateTime createdAt) implements Serializable {

    public static Post of(String title, String content) {
        return new Post(null, title, content, LocalDateTime.now());
    }
}

File: mongodb-panache-kotlin/src/main/kotlin/com/example/repository/Post.kt
package com.example.repository

import org.bson.types.ObjectId
import java.time.LocalDateTime

data class Post(
    var id: ObjectId? = null,// used by MongoDB for the _id field var
    var title: String,
    var body: String,
    var createdAt: LocalDateTime? = LocalDateTime.now()
)
